### ISSUE TITLE
[Enhancement] make the send button unclickable when CLoudflare Turnstile has not verified you

### ISSUE THREAD

**Issue by PythonSmall-Q** [langningchen/XMOJ-Script#58]
Maybe consider changing cloudflare turnstile to beta or using the paid plan since cloudflare turnstile is slow in some regions for example Tibet and Xinjiang where i am now travelling in
also, why not make the send button unclickable when turnstile has not verified you, also changing turnstile setting to auto mode(which only show when needed which saves a lot of time)

**Comment by PythonSmall-Q** [comment]
@langningchen @boomzero have a look

**Comment by PythonSmall-Q** [comment]
auto mode can be seen at xmoj.flarum.cloud
the register button has cloudflare turnstile set to auto mode

**Comment by PythonSmall-Q** [comment]
Talking about that ddos thing, i searched in ip2location and cloudflare, maybe you can consider enabling the "超级机器人防御" in WFA

**Comment by PythonSmall-Q** [comment]
that can solve most ddos or you can upgrade to paid plan to get better safety enhancement

**Comment by PythonSmall-Q** [comment]
plz relpy in chinese. i am writing in english to bypass xinjiang GFW

**Comment by boomzero** [comment]
> Talking about that ddos thing, i searched in ip2location and cloudflare, maybe you can consider enabling the "超级机器人防御" in WFA

It's on the paid plan, I think

**Comment by boomzero** [comment]
![](https://github.com/langningchen/XMOJ-Script/assets/85378277/1be4e94a-fb9a-4fd8-bdd2-f8a8531c34af)

**Comment by boomzero** [comment]
> Maybe consider changing cloudflare turnstile to beta or using the paid plan since cloudflare turnstile is slow in some regions for example Tibet and Xinjiang where i am now travelling in also, why not make the send button unclickable when turnstile has not verified you, also changing turnstile setting to auto mode(which only show when needed which saves a lot of time)

It's currently in non-interactive mode.

**Comment by boomzero** [comment]
![](https://github.com/langningchen/XMOJ-Script/assets/85378277/8a96a790-d5ff-4aa7-a5ed-1364e8d2df47)
I'm not sure what you meant by auto.

**Comment by boomzero** [comment]
> that can solve most ddos or you can upgrade to paid plan to get better safety enhancement

https://github.com/langningchen/XMOJ-Script/issues/35#issuecomment-1679985888

**Comment by boomzero** [comment]
> why not make the send button unclickable when turnstile has not verified you

我看一下

**Comment by boomzero** [comment]
> > Talking about that ddos thing, i searched in ip2location and cloudflare, maybe you can consider enabling the "超级机器人防御" in WFA
> 
> It's on the paid plan, I think

哦, 对, userscript也是bot, 也会被block

**Comment by langningchen** [comment]
好吧

**Comment by langningchen** [comment]
那么这个Issue就Close掉？

**Comment by boomzero** [comment]
>   why not make the send button unclickable when turnstile has not verified you

这个可以做

**Comment by boomzero** [comment]
关于speed, Cloudflare turnstile 和 xmoj-bbs 都host在cloudflare cdn上, 如果你一个打不开那另一个也打不开

**Comment by langningchen** [comment]
哦，好的

**Comment by boomzero** [comment]
I don't think this is necessary, you'll get an error if you try to submit before the captcha completes anyway

**Comment by langningchen** [comment]
但是我认为有必要

**Comment by langningchen** [comment]
这样可以减少给服务器的无效请求以降低服务器负担

**Comment by langningchen** [comment]
OK

**Comment by PythonSmall-Q** [comment]
是managed模式 @boomzero @langningchen

**Comment by langningchen** [comment]
@boomzero

**Comment by boomzero** [comment]
?

**Comment by langningchen** [comment]
怎么了

**Comment by langningchen** [comment]
你看看有没有`managed模式`

**Comment by boomzero** [comment]
> 是managed模式 @boomzero @langningchen

Managed是要点的啊

**Comment by boomzero** [comment]
就是要Click的模式

**Comment by langningchen** [comment]
不知道，你 @PythonSmall-Q 问问

**Comment by boomzero** [comment]
> ![](https://user-images.githubusercontent.com/85378277/262825360-8a96a790-d5ff-4aa7-a5ed-1364e8d2df47.jpg) I'm not sure what you meant by auto.

是的

**Comment by PythonSmall-Q** [comment]
> 关于speed, Cloudflare turnstile 和 xmoj-bbs 都host在cloudflare cdn上, 如果你一个打不开那另一个也打不开

有误，cf的turnstile中国点在SHANGHAI和GUIZHOU的Datacentre。但是xmojbbs的host其实在美国
